addons/mail/static/tests/chatter/web/chatter.test.js

```diff
@@ -147,13 +147,12 @@ test("No attachment loading spinner when creating records", async () => {
 });
 
 test("No attachment loading spinner when switching from loading record to creation of record", async () => {
-    onRpc("/mail/data", async (request) => {
-        const { params } = await request.json();
-        if (params.fetch_params.some((fetchParam) => fetchParam[0] === "mail.thread")) {
-            await new Deferred();
-        }
-    });
+    const def = new Deferred();
     const pyEnv = await startServer();
+    const { waitStoreFetch } = listenStoreFetch("mail.thread", async () => {
+        asyncStep("before mail.thread");
```

Comment:
Note: this step is necessary to ensure the callback is actually applied, it makes the test much more robust.